src/main/java/org/apache/sling/auth/oauth_client/impl/OidcConnectionImpl.java

```diff
@@ -57,11 +67,56 @@ String webconsole_configurationFactory_nameHint() default
 
     private final Config cfg;
     private final OidcProviderMetadataRegistry metadataRegistry;
+    private final String authorizationEndpoint;
+    private final String tokenEndpoint;
+    private final String userInfoUrl;
+    private final String jwkSetURL;
+    private final String issuer;
+    private final boolean hasBaseUrl;
 
     @Activate
     public OidcConnectionImpl(Config cfg, @Reference OidcProviderMetadataRegistry metadataRegistry) {
         this.cfg = cfg;
         this.metadataRegistry = metadataRegistry;
+        this.authorizationEndpoint = cfg.authorizationEndpoint();
+        this.tokenEndpoint = cfg.tokenEndpoint();
+        this.userInfoUrl = cfg.userInfoUrl();
+        this.jwkSetURL = cfg.jwkSetURL();
+        this.issuer = cfg.issuer();
+
+        // Validate configuration: either baseUrl is provided OR all explicit endpoints are provided
+        hasBaseUrl = !isNullOrEmpty(cfg.baseUrl());
+
+        if (hasBaseUrl
+                && !isNullOrEmpty(tokenEndpoint)
+                && !isNullOrEmpty(authorizationEndpoint)
+                && !isNullOrEmpty(userInfoUrl)
+                && !isNullOrEmpty(jwkSetURL)
+                && !isNullOrEmpty(issuer)) {
+            throw new IllegalArgumentException(
+                    "Either baseUrl OR explicit endpoints "
+                            + "(authorizationEndpoint, tokenEndpoint, userInfoUrl, jwkSetURL, issuer) must be provided, not both");
+        }
+
+        if (!hasBaseUrl
+                && isNullOrEmpty(tokenEndpoint)
+                && isNullOrEmpty(authorizationEndpoint)
+                && isNullOrEmpty(userInfoUrl)
+                && isNullOrEmpty(jwkSetURL)
+                && isNullOrEmpty(issuer)) {
```

Comment:
i believe my concern is addressed but i am not 100% sure because i have a hard time processing the long ifs :)

@nscendoni , would it make sense to extract the check for an explicit-config into a seprate method? i find the condition here super hard to process.... in combination with the other if above, i had to look twice if this is just a basic duplication or not.